logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/RegisterClientUseCase.kt

```diff
@@ -0,0 +1,28 @@
+package com.wire.kalium.logic.feature.client
+
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.client.RegisterClientParam
+import com.wire.kalium.logic.failure.AuthenticationFailure
+import com.wire.kalium.logic.functional.Either
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+
+class RegisterClientUseCase(
+    private val clientRepository: ClientRepository
+) {
+    suspend operator fun invoke(param: RegisterClientParam): Flow<RegisterClientResult> = flow {
+        emit(RegisterClientResult.Loading)
```

Comment:
I'd say that `Loading` is not a result, it doesn't add much information to whoever is using, and it may need customization by whoever is using it if multiple flows are used:

```kotlin
fooUseCase().collect {
    // Handle possible progress bar, but try not to interfere with the other usecase
}
```

Could be:
```kotlin
showProgressBar()
fooUseCase().collect{}
hideProgressBar()
```

----

On more complicated cases it can get messed up:
```kotlin
fooUseCase().collect {
    // Handle possible progress bar, but try not to interfere with the other usecase
}

barUseCase().collect {
    // Handle possible progress bar, but try not to interfere with the other usecase
}
```

And we could solve by adding the jobs to another supervisor job and changing a LiveData as they finish: 

```kotlin
class UserProfileViewModel {

    private val isLoading = MutableLiveData<Boolean>(false)
    private val loadingJob = SupervisorJob()
    private val loadingScope = CoroutineScope(Dispatchers.IO + loadingJob) //TODO: Handle cancellation on destroy

    private fun loadingAction(action: (suspend CoroutineScope.() -> Unit)) {
        val newJob = loadingScope.launch { action() }
        uploadLoadingStatus() // Immediately update the LoadingStatus
        newjob.launchOnCompletion { updateLoadingStatus() } // Update again when completing this job
    }

    /**
     * Checks if there is any loading job that is still running updating the [isLoading] LiveData accordingly
     */
    private fun updateLoadingStatus() {
        isLoading.postValue(
            loadingJob.children.any { it.isActive }
        )
    }

    fun loadUserProfile() {
        // Both actions run in parallel:
        loadingAction {
            getUserProfileData().collect {
                // handle result only. No need to handle loading state
            }
        }
        loadingAction {
            getUserClientData().collect {
                // handle result only. No need to handle loading state
            }
        }
    }
}
```



TL;DR: Maybe we could leave loading status to whoever is using this library to figure out, not us :shrug: ? What do you think?

Reply:
make sense, i will do the changes